karapace/protobuf/proto_normalizations.py

```diff
@@ -93,12 +93,17 @@ class NormalizedOneOfElement(OneOfElement):
     groups: Sequence[NormalizedGroupElement]
 
 
-def type_field_element_with_sorted_options(type_field: FieldElement) -> NormalizedFieldElement:
+def type_field_element_with_sorted_options(type_field: FieldElement, scope: str) -> NormalizedFieldElement:
     sorted_options = None if type_field.options is None else list(sorted(type_field.options, key=sort_by_name))
+    field_type_normalized = type_field.element_type.removeprefix(
+        "."
+    ).removeprefix(  # getting rid of the optional `.` before the package name
```

Comment:
all the fully qualified references always start with a `.`, like for the example of: `.google.protobuf.Timestamp` that its referring to the package `google.protobuf`.
In this case normalizing it by removing the `.` can lead to shadowing if the import of the google message its after another import that defines a scope that contains google, like for e.g.:

```
package my.awesome.tricky.company.google.protobuf;

message Timestamp {
   // definition...
}
```

Even if this is possible its unlikely to happen, we may even reduce that removal of the `.` just for types that do not have other anchestors in type type trie

Reply:
Implemented it by the type tree and implemented a test